cli/args.go

```diff
@@ -360,15 +366,31 @@ func parseIAMRoleOptions(args []string) (*options.IAMRoleOptions, error) {
 	return optsOut, nil
 }
 
+// getOptionArgIfMatched returns the value to use for the option key, if the arg matches the option. This will handle
+// options passed in as --OPTION_KEY VALUE or --OPTION_KEY=VALUE. The second boolean return value indicates if the arg
+// matches the option, returning false if it does not. This will error if it expects a value in the next arg, but the
+// current arg is the end of the input.
+func getOptionArgIfMatched(arg string, nextArg *string, optionName string) (string, bool, error) {
+	optionOnly := fmt.Sprintf("--%s", optionName)
+	if arg == optionOnly {
+		logIfDeprecatedOption(optionName)
+		if nextArg == nil {
+			return "", true, errors.WithStackTrace(ArgMissingValue(optionName))
+		}
+		return *nextArg, true, nil
+	} else if strings.HasPrefix(arg, optionOnly+"=") {
+		logIfDeprecatedOption(optionName)
```

Comment:
NIT: consider moving the `logIfDeprecatedOption(optionName)` to above the conditional.